contracts/DutchExchange.sol

```diff
@@ -1101,6 +1101,121 @@ contract DutchExchange {
+        address[] tokens
+    )
+        public
+        view
+        returns (address[] tokens1, address[] tokens2)
+    {
+        uint arrayLength;
+
+        for (uint k = 0; k < tokens.length - 1; k++) {
+            for (uint l = k + 1; l < tokens.length; l++) {
+                getAuctionIndex(tokens[k], tokens[l]) > 0 ? arrayLength++ : 0;
+            }
+        }
+
+        tokens1 = new address[](arrayLength);
+        tokens2 = new address[](arrayLength);
+
+        uint h;
+
+        for (uint i = 0; i < tokens.length - 1; i++) {
+            for (uint j = i + 1; j < tokens.length; j++) {
+                if (getAuctionIndex(tokens[i], tokens[j]) > 0) {
+                    tokens1[h] = tokens[i];
+                    tokens2[h] = tokens[j];
+                    h++;
+                }
+            }
+        }
+    }
+    
+    //@dev for quick overview of possible sellerBalances to calculate the possible withdraw tokens
+    //@param auctionSellToken is the sellToken defining an auctionPair
+    //@param auctionBuyToken is the buyToken defining an auctionPair
+    //@param user is the user who wants to his tokens
+    //@param lastNAuctions how many auctions will be checked. 0 means all
+    //@returns returns sellbal for all indices for all tokenpairs 
+    function getIndicesWithClaimableTokens(
+        address auctionSellToken,
+        address auctionBuyToken,
+        address user,
+        uint lastNAuctions
+    )
+        public
+        view
+        returns(uint[] indices, uint[] balances)
+    {
+        uint runningAuctionIndex = getAuctionIndex(auctionSellToken, auctionBuyToken);
+
+        uint arrayLength;
+        
+        uint startingIndex = lastNAuctions == 0 ? 1 : runningAuctionIndex - lastNAuctions + 1;
+
+        for (uint j = startingIndex; j <= runningAuctionIndex; j++) {
+            sellerBalances[auctionSellToken][auctionBuyToken][j][user] > 0 ? arrayLength++ : 0;
+        }
+
+        indices = new uint[](arrayLength);
+        balances = new uint[](arrayLength);
+
+        uint k;
+
+        for (uint i = 1; i <= runningAuctionIndex; i++) {
+            if (sellerBalances[auctionSellToken][auctionBuyToken][i][user] > 0) {
+                indices[k] = i;
+                balances[k] = sellerBalances[auctionSellToken][auctionBuyToken][i][user];
+                k++;
+            }
+        }
+    }    
+
+    //@dev for quick overview of current sellerBalances for a user
+    //@param auctionSellTokens are the sellTokens defining an auctionPair
+    //@param auctionBuyTokens are the buyTokens defining an auctionPair
+    //@param user is the user who wants to his tokens
+    function getSellerBalancesOfCurrentAuctions(
+        address[] auctionSellTokens,
+        address[] auctionBuyTokens,
+        address user
+    )
+        public
+        view
+        returns (uint[])
+    {
+        uint length = auctionSellTokens.length;
+        uint length2 = auctionBuyTokens.length;
```

Comment:
would setting `length2` here back to 0 save gas? or am I being dumb 😓

Reply:
hopefully we will never have to pay any gas for the constant view functions :)